RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  ManageSubscriptionsViewModel.swift
+//
+//
+//  Created by Cesar de la Vega on 27/5/24.
+//
+
+import Foundation
+import RevenueCat
+
+@available(iOS 15.0, *)
+class ManageSubscriptionsViewModel: ObservableObject {
+
+    var isLoaded: Bool {
+        if case .notLoaded = state {
+            return false
+        }
+        return true
+    }
+
+    @Published
+    var subscriptionInformation: SubscriptionInformation?
+
+    @Published
+    var refundRequestStatus: String?
+
+    @Published
+    var configuration: CustomerCenterData?
+    var error: Error?
+
+    enum State {
+
+        case notLoaded
+        case success
+        case error(Error)
+
+    }
+
+    private(set) var state: State {
+        didSet {
+            if case let .error(stateError) = state {
+                self.error = stateError
+            }
+        }
+    }
+
+    init() {
+        state = .notLoaded
+    }
+
+    init(configuration: CustomerCenterData) {
+        state = .notLoaded
+        self.configuration = configuration
+    }
+
+    init(configuration: CustomerCenterData, subscriptionInformation: SubscriptionInformation) {
+        self.configuration = configuration
+        self.subscriptionInformation = subscriptionInformation
+        state = .success
+    }
+
+    func loadSubscriptionInformation() async {
+        do {
+            let customerInfo = try await Purchases.shared.customerInfo()
+            guard let currentEntitlementDict = customerInfo.entitlements.active.first,
+                  let subscribedProductID = customerInfo.activeSubscriptions.first,
+                  let subscribedProduct = await Purchases.shared.products([subscribedProductID]).first else {
+                return
+            }
+            let currentEntitlement = currentEntitlementDict.value
+
+            self.subscriptionInformation = SubscriptionInformation(
+                title: subscribedProduct.localizedTitle,
+                duration: subscribedProduct.subscriptionPeriod?.durationTitle ?? "",
+                price: subscribedProduct.localizedPriceString,
+                nextRenewal: "\(String(describing: currentEntitlement.expirationDate!))",
```

Comment:
Lifetime access will not be part of the `activeSubscriptions`, because that would be a non-subscription product. We should add support for non-subscriptions though. I am going to create a task for that. For now, I think this code is fine